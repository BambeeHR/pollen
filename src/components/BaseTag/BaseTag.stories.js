import { boolean, select, text } from '@storybook/addon-knobs';
import BaseTag, { Sizes } from './BaseTag.vue';
import { extendedPalette } from '../../tailwind/theme';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

export default {
  title: 'BaseTag',
  component: BaseTag,
  parameters: {
    componentSubtitle: 'Basic tag component.',
  },
};

const colors = ['black', 'gray', ...Object.keys(extendedPalette)];

export const Gallery = () => ({
  components: { BaseTag, TypeOverline },
  data() {
    return {
      sizes: Sizes,
      colors,
    };
  },
  template: `
    <div>
      <div class="flex">
        <div class="w-1/2">
          <TypeOverline tag="h2" class="mb-2" variant="large">Filled</TypeOverline>
          <div v-for="color in colors" :key="color">
            <TypeOverline tag="h3" class="mb-2">{{ color }}</TypeOverline>
            <div class="flex items-end mb-2"> 
              <div v-for="size in sizes" :key="'filled-' + color + size" class="mr-8">
                <BaseTag :color="color" :size="size">{{ size }}</BaseTag>
              </div>
            </div>
          </div>
        </div>
        <div class="w-1/2">
          <TypeOverline tag="h2" class="mb-2" variant="large">Outline</TypeOverline>
          <div v-for="color in colors" :key="color">
            <TypeOverline tag="h3" class="mb-2">{{ color }}</TypeOverline>
            <div class="flex items-end mb-2"> 
              <div v-for= "size in sizes" :key="'outline-' + color + size" class="mr-8">
                <BaseTag :color="color" :size="size" outline>{{ size }}</BaseTag>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  `,
});

export const WithKnobs = () => {
  return {
    components: { BaseTag },
    props: {
      text: {
        default: text('Text', 'Hello world'),
      },
      color: {
        default: select('Color', colors, colors[0]),
      },
      size: {
        default: select('Size', Object.values(Sizes), Sizes.MEDIUM),
      },
      outline: {
        default: boolean('Outline', false),
      },
    },
    template: `
      <BaseTag
        :color="color"
        :size="size"
        :outline="outline"
      >
        {{ text }}
      </BaseTag>
    `,
  };
};

/* eslint-disable no-param-reassign */
[Gallery, WithKnobs].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['BaseTag.test.js'],
    },
  };
});
