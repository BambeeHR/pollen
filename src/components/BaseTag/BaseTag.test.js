import { mount } from '@vue/test-utils';
import BaseTag from './BaseTag.vue';
import { Gallery } from './BaseTag.stories';

describe('BaseTag', () => {
  test('renders correctly', () => {
    const wrapper = mount(BaseTag, {
      slots: {
        default: 'Hello world',
      },
      propsData: {
        color: 'inevitable',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders gallery correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });
});
