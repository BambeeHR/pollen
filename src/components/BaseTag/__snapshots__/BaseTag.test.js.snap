// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTag renders correctly 1`] = `
<div
  class="base-tag base-tag--inevitable base-tag--medium base-tag--filled"
>
  Hello world
</div>
`;

exports[`BaseTag renders gallery correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="w-1/2"
    >
      <h2
        class="type-overline type-overline-large mb-2"
      >
        Filled
      </h2>
       
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          geyser
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          tuberose
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          gingerrr
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          rose
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          tangerine
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          maroon
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          puce
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          inevitable
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          virginia
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          mosaic
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          eastern
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          caribbean
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          evergreen
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--small base-tag--filled"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--medium base-tag--filled"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--large base-tag--filled"
            >
              large
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="w-1/2"
    >
      <h2
        class="type-overline type-overline-large mb-2"
      >
        Outline
      </h2>
       
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          geyser
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--geyser base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          tuberose
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tuberose base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          gingerrr
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--gingerrr base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          rose
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--rose base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          tangerine
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--tangerine base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          maroon
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--maroon base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          puce
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--puce base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          inevitable
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--inevitable base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          virginia
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--virginia base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          mosaic
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--mosaic base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          eastern
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--eastern base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          caribbean
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--caribbean base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="type-overline type-overline-small mb-2"
        >
          evergreen
        </h3>
         
        <div
          class="flex items-end mb-2"
        >
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--small base-tag--outline"
            >
              small
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--medium base-tag--outline"
            >
              medium
            </div>
          </div>
          <div
            class="mr-8"
          >
            <div
              class="base-tag base-tag--evergreen base-tag--large base-tag--outline"
            >
              large
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
